src/scanpy/preprocessing/_simple.py

```diff
@@ -722,13 +736,13 @@ def regress_out(
                 "we regress on the mean for each category."
             )
         logg.debug("... regressing on per-gene means within categories")
-        regressors = np.zeros(X.shape, dtype="float32")
+        # Create numpy array's from categorical variable
+        number_categories = np.int64(len(adata.obs[keys[0]].cat.categories))
+        filters = adata.obs[keys[0]].cat.codes.to_numpy()
+        number_categories = number_categories.astype(filters.dtype)
```

Comment:
Either this or add a comment (to the code) explaining why it needs to be the other way

```suggestion
        number_categories = len(adata.obs[keys[0]].cat.categories)
        filters = adata.obs[keys[0]].cat.codes.to_numpy()
```

Reply:
> Other wise you have a dtype missmatch and crash of the kernel

I would say that this is the important part for the comment!

Reply:
Also the way to do this in one step is

```suggestion
        filters = adata.obs[keys[0]].cat.codes.to_numpy()
        number_categories = filters.dtype.type(len(adata.obs[keys[0]].cat.categories))
```